server/game/cards/03-WC/Memolith.js

```diff
@@ -0,0 +1,39 @@
+const Card = require('../../Card.js');
+
+class Memolith extends Card {
+    setupCardAbilities(ability) {
+        this.action({
+            targets: {
+                graftOrTrigger: {
+                    mode: 'select',
+                    choices: {
+                        'Graft an action card': (context) => context.player.hand.filter(card => card.type === 'action'),
+                        'Trigger play effect': (context) => context.source.childCards.some(card => !card.facedown)
+                    }
+                },
+                action: {
+                    location: 'any',
+                    controller: 'self',
+                    dependsOn: 'graftOrTrigger',
+                    cardCondition: (card, context) => context.selects.graftOrTrigger &&
+                        (((context.selects.graftOrTrigger.choice === 'Graft an action card') && (card.location === 'hand') && (card.type === 'action'))
+                            || ((context.selects.graftOrTrigger.choice === 'Trigger play effect') && (card.parent === this) && (!card.facedown))),
+                    gameAction: [
```

Comment:
this looks like a good place for a conditional action?

Reply:
Fixed it using ConditionalAction